lib/tests/2wp.js

```diff
@@ -29,6 +32,87 @@ const setupBridgeDonation = async (rskTxHelpers, btcTxHelper) => {
+    if(change > 0) {
+      tx.addOutput(
+        bitcoinJsLib.address.toOutputScript(sendersInfo[index].btcSenderAddressInfo.address, btcTxHelper.btcConfig.network),
+        Number(btcToSatoshis(change))
+      );
+    }
+  });
+};
+
+const addOutputsToFed = (tx, outputsToFed) => {
+  outputsToFed.forEach(outputAmount => {
+    tx.addOutput(
+      bitcoinJsLib.address.toOutputScript(federationAddress, btcTxHelper.btcConfig.network),
+      Number(btcToSatoshis(outputAmount))
+    );
+  });
+};
+
+const pushPegin = async (btcPeginTxHash, expectedUtxosCount) => {
+  await waitForBitcoinTxToBeInMempool(btcTxHelper, btcPeginTxHash);
+  await mineForPeginRegistration(rskTxHelper, btcTxHelper);
+  await ensurePeginIsRegistered(rskTxHelper, btcPeginTxHash, expectedUtxosCount);
+};
+
+const getSendersInfo = async (initialBtcSenderBalancesInBtc) => {
+  return await Promise.all(initialBtcSenderBalancesInBtc.map(initialBtcSenderBalance => createSenderRecipientInfo(rskTxHelper, btcTxHelper, 'legacy', initialBtcSenderBalance + btcTxHelper.getFee())));
+};
+
+const getSendersBtcAddressBalances = async (sendersInfo) => {
+  return await Promise.all(sendersInfo.map(senderInfo => btcTxHelper.getAddressBalance(senderInfo.btcSenderAddressInfo.address)));
+};
+
+const getSendersUtxosInfo = async (sendersInfo, btcSenderAmountsToSendToFed) => {
+  return await Promise.all(sendersInfo.map((senderInfo, index) => btcTxHelper.selectSpendableUTXOsFromAddress(senderInfo.btcSenderAddressInfo.address, btcSenderAmountsToSendToFed[index])));
+};
+
+const getSendersChange = (sendersUtxosInfo) => {
+  return sendersUtxosInfo.map(senderUtxosInfo => senderUtxosInfo.change - btcTxHelper.getFee());
+};
+
+const getSendersPrivateKeys = (sendersInfo) => {
+  return sendersInfo.map(senderInfo => senderInfo.btcSenderAddressInfo.privateKey);
+};
+
+const getTotalAmountToFed = (outputsToFed) => {
+  return outputsToFed.reduce((total, amount) => total + amount, 0);
+};
+
+const getFinalRskRecipientBalances = async (sendersInfo) => {
+  return await Promise.all(sendersInfo.map(senderInfo => rskTxHelper.getBalance(senderInfo.rskRecipientRskAddressInfo.address)));
+};
+
+const legacyPeginTestsHappyPath = [
+  {
+    description: 'should do legacy pegin with one input and one output to the federation',
+    initialBtcSenderBalancesInBtc: [ MINIMUM_PEGIN_VALUE_IN_BTC ],
+    btcSenderAmountsToSendToFed: [ MINIMUM_PEGIN_VALUE_IN_BTC ],
+    outputsToFed: [ MINIMUM_PEGIN_VALUE_IN_BTC ],
+  },
+  {
+    description: 'should do legacy pegin with multiple inputs from different accounts and one output to the federation',
+    initialBtcSenderBalancesInBtc: [ MINIMUM_PEGIN_VALUE_IN_BTC, MINIMUM_PEGIN_VALUE_IN_BTC, MINIMUM_PEGIN_VALUE_IN_BTC ], // Each sender is funded with some amount
+    btcSenderAmountsToSendToFed: [ MINIMUM_PEGIN_VALUE_IN_BTC, MINIMUM_PEGIN_VALUE_IN_BTC, MINIMUM_PEGIN_VALUE_IN_BTC ], // Each sender decides how much to send to the federation
```

Comment:
`btcSenderAmountsToSendToFed` seems a bit complex name to me. Maybe `peginSenderAmounts` by including pegin, we can assume this refers to funds sent to fed. Wdyt?